sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -684,6 +685,24 @@ object TypeCoercion {
     }
   }
 
+  /**
+   * Since SPARK-16323 `IntegralDivide` always returns long-type value.
```

Comment:
This is a bit misleading. The `DIV` operator always returns long-type values since the first day. SPARK-16323 is more like an internal perf optimization.

How about
```
The DIV operator always returns long-type value. This rule cast ...
```